util/Migrator/DbScripts/2022-07-11_00_Initial_Organization.sql

```diff
@@ -0,0 +1,547 @@
+IF OBJECT_ID('[dbo].[OrganizationPasswordManager]') IS NOT NULL
+
+BEGIN
+    DROP TABLE [dbo].[OrganizationPasswordManager]
+END
+GO
+
+SET ANSI_NULLS ON
+GO
+SET QUOTED_IDENTIFIER ON
+GO
+CREATE TABLE [dbo].[OrganizationPasswordManager](
+    [Id] [uniqueidentifier] NOT NULL,
+    [OrganizationId] [uniqueidentifier] NOT NULL,
+    [Plan] [nvarchar](50) NOT NULL,
+    [PlanType] [tinyint] NOT NULL,
+    [Seats] [int] NULL,
+    [MaxCollections] [smallint] NULL,
+    [UseTotp] [bit] NULL,
+    [UsersGetPremium] [bit] NOT NULL,
+    [Storage] [bigint] NULL,
+    [MaxStorageGb] [smallint] NULL,
+    [MaxAutoscaleSeats] [int] NULL,
+    [RevisionDate] DATETIME NULL
+) 
+GO
+
+IF OBJECT_ID('[dbo].[OrganizationSecretsManager]') IS NOT NULL
+
+BEGIN
+    DROP TABLE [dbo].[OrganizationSecretsManager]
+END
+GO
+
+CREATE TABLE [dbo].[OrganizationSecretsManager](
+    [Id] [uniqueidentifier] NOT NULL,
+    [OrganizationId] [uniqueidentifier] NOT NULL,
+    [Plan] [nvarchar](50) NOT NULL,
+    [PlanType] [tinyint] NOT NULL,
+    [UserSeats] [int] NULL,
+    [ServiceAccountSeats] [int] NULL,
+    [UseEnvironments] [bit] NULL,
+    [MaxAutoscaleUserSeats] [int] NULL,
+    [MaxAutoscaleServiceAccounts] [int] NULL,
+    [MaxProjects] [int] NULL,
+    [RevisionDate] DATETIME NULL
+) 
+GO
+
+
+IF OBJECT_ID('[dbo].[Organization_DeleteById]') IS NOT NULL
+
+BEGIN
+    DROP PROCEDURE [dbo].[Organization_DeleteById]
+
+END
+GO
+
+
+SET ANSI_NULLS ON
+GO
+SET QUOTED_IDENTIFIER ON
+GO
+
+CREATE PROCEDURE [dbo].[Organization_DeleteById]
+    @Id UNIQUEIDENTIFIER
+AS
+BEGIN
+    SET NOCOUNT ON
+
+    EXEC [dbo].[User_BumpAccountRevisionDateByOrganizationId] @Id
+
+    DECLARE @BatchSize INT = 100
+    WHILE @BatchSize > 0
+    BEGIN
+        BEGIN TRANSACTION Organization_DeleteById_Ciphers
+
+        DELETE TOP(@BatchSize)
+        FROM
+            [dbo].[Cipher]
+        WHERE
+            [UserId] IS NULL
+            AND [OrganizationId] = @Id
+
+        SET @BatchSize = @@ROWCOUNT
+
+        COMMIT TRANSACTION Organization_DeleteById_Ciphers
+    END
+
+    BEGIN TRANSACTION Organization_DeleteById
+
+    DELETE
+    FROM
+        [dbo].[SsoUser]
+    WHERE
+        [OrganizationId] = @Id
+
+    DELETE
+    FROM
+        [dbo].[SsoConfig]
+    WHERE
+        [OrganizationId] = @Id
+
+    DELETE CU
+    FROM 
+        [dbo].[CollectionUser] CU
+    INNER JOIN 
+        [dbo].[OrganizationUser] OU ON [CU].[OrganizationUserId] = [OU].[Id]
+    WHERE 
+        [OU].[OrganizationId] = @Id
+
+    DELETE
+    FROM 
+        [dbo].[OrganizationUser]
+    WHERE 
+        [OrganizationId] = @Id
+
+    DELETE
+    FROM
+         [dbo].[ProviderOrganization]
+    WHERE
+        [OrganizationId] = @Id
+
+    EXEC [dbo].[OrganizationApiKey_OrganizationDeleted] @Id
+    EXEC [dbo].[OrganizationConnection_OrganizationDeleted] @Id
+    EXEC [dbo].[OrganizationSponsorship_OrganizationDeleted] @Id
+
+    DELETE
+    FROM
+        [dbo].[Organization]
+    WHERE
+        [Id] = @Id
```

Comment:
This needs to be last. Since the two tables below should always have an organization. (And we probably need to make update their table to have a non nullable field for the relationship + ensure it's valid)